crates/cxx-qt-lib/src/qml/qqmlapplicationengine.rs

```diff
@@ -101,4 +124,21 @@ impl QQmlApplicationEngine {
     pub fn new() -> cxx::UniquePtr<Self> {
         ffi::qqmlapplicationengine_new()
     }
+
+    /// Returns the instance of a singleton type named typeName from the module specified by uri.
+    /// This function was introduced in Qt 6.5.
+    #[cfg(any(cxxqt_qt_version_at_least_7, cxxqt_qt_version_at_least_6_5))]
+    pub fn singleton_instance<T>(
+        self: Pin<&mut Self>,
+        uri: QAnyStringView,
+        type_name: QAnyStringView,
+    ) -> Option<&mut T> {
+        unsafe {
+            let ptr = ffi::qqmlapplicationengine_singleton_instance(self, uri, type_name);
+            if ptr.is_null() {
+                return None;
+            }
+            Some(&mut *(ptr as *mut T))
```

Comment:
Hm, this is definitely *not* safe to do here!

As you could basically "spoof" T to be different from the actual type that the singleton has.

In the most extreme case, I could just do:
```
engine.singleton_instance::<i32>(...);
```

We could mark the function itself as `unsafe` to move that responsibility up one level, but it's still not great...

Reply:
I have added some lifetime checks to improve it a bit, but yes this is inherently unsafe due to a lack of type-checking. Note that once we _have_ the ability to do type checking (like in #562, or akiselev's comment), this should stay unsafe.

Reply:
I'll leave it up to your discretion whether we should include this inherently-unsafe version now, and then have a proper solution in tandem with #562. IMO, it would be really useful right now :-)